src/libstd/sys/windows/fs.rs

```diff
@@ -333,6 +334,91 @@ impl File {
         Ok(newpos as u64)
     }
 
+    pub fn remove(&self) -> io::Result<()> {
+        let mut info = c::FILE_DISPOSITION_INFO {
+            DeleteFile: -1,
+        };
+        let size = mem::size_of_val(&info);
+        try!(cvt(unsafe {
+            c::SetFileInformationByHandle(self.handle.raw(),
+                                          c::FileDispositionInfo,
+                                          &mut info as *mut _ as *mut _,
+                                          size as c::DWORD)
+        }));
+        Ok(())
+    }
+
+    pub fn rename(&self, new: &Path, replace: bool) -> io::Result<()> {
+        // &self must be opened with DELETE permission
+
+        #[cfg(target_arch = "x86")]
+        const STRUCT_SIZE: usize = 12;
+        #[cfg(target_arch = "x86_64")]
+        const STRUCT_SIZE: usize = 20;
+
+        // FIXME: check for internal NULs in 'new'
+        let reserved = OsStr::new(&"\0\0\0\0\0\0\0\0\0\0"[..STRUCT_SIZE/2]);
+        let mut data: Vec<u16> = reserved.encode_wide()
```

Comment:
Yes, much easier :). But I could not get the iterator to chain with the next, which gives some `EncodeWide` type. This is the best I could come up with, but I am happy to change it.